api/src/routes/api.ts

```diff
@@ -7,10 +7,13 @@ import Job from "../models/Job";
 
 const router = express.Router();
 import { setupPassport, requireToken } from "./auth";
+import fs = require("fs");
 setupPassport(router);
 
 function getJobFromRequest(req): Job {
-    // TODO: sanitize parameters!
+    if (!fs.existsSync(holdingArea() + req.params.category + "/" + req.params.job)) {
+        return null; //res.status(404).json({ error: "Not Found"});
```

Comment:
All of the code calling getJobFromRequest needs to be able to handle the possibility that it will return null, and return a 404 when it does. Right now it looks like everything just assumes this will work, which is a problem.